sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcFileFormat.scala

```diff
@@ -167,8 +169,10 @@ class OrcFileFormat
         val iter = new RecordReaderIterator[OrcStruct](orcRecordReader)
         Option(TaskContext.get()).foreach(_.addTaskCompletionListener(_ => iter.close()))
 
-        val unsafeProjection = UnsafeProjection.create(requiredSchema)
-        val deserializer = new OrcDeserializer(dataSchema, requiredSchema, requestedColIds)
+        val colIds = requestedColIds ++ List.fill(partitionSchema.length)(-1).toArray[Int]
+        val unsafeProjection = UnsafeProjection.create(resultSchema)
```

Comment:
can we follow parquet and just join the data row and partition row, and do a final unsafe projection? It's much easier and there is no performance difference.

Reply:
Parquet Vectorization work like the following.
```
      // UnsafeRowParquetRecordReader appends the columns internally to avoid another copy.
      if (parquetReader.isInstanceOf[VectorizedParquetRecordReader] &&
          enableVectorizedReader) {
        iter.asInstanceOf[Iterator[InternalRow]]
```

Reply:
Oh, I see. you meant non-vectorized path. Sorry, I was confused since I focused too much on vectorized path. I'll do.